compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmUnionTypeConstantsGen.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.ballerinalang.compiler.bir.codegen;
+
+import org.ballerinalang.model.elements.PackageID;
+import org.objectweb.asm.ClassWriter;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BType;
+import org.wso2.ballerinalang.compiler.semantics.model.types.BUnionType;
+
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+import static org.objectweb.asm.ClassWriter.COMPUTE_FRAMES;
+import static org.objectweb.asm.Opcodes.ACC_FINAL;
+import static org.objectweb.asm.Opcodes.ACC_PRIVATE;
+import static org.objectweb.asm.Opcodes.ACC_PUBLIC;
+import static org.objectweb.asm.Opcodes.ACC_STATIC;
+import static org.objectweb.asm.Opcodes.ACC_SUPER;
+import static org.objectweb.asm.Opcodes.ALOAD;
+import static org.objectweb.asm.Opcodes.GETSTATIC;
+import static org.objectweb.asm.Opcodes.INVOKESPECIAL;
+import static org.objectweb.asm.Opcodes.INVOKESTATIC;
+import static org.objectweb.asm.Opcodes.RETURN;
+import static org.objectweb.asm.Opcodes.V1_8;
+
+/**
+ * Generates Jvm class for the ballerina union types as constants for a given module.
+ *
+ * @since 2.0.0
+ */
+public class JvmUnionTypeConstantsGen {
+
+    private final Map<BUnionType, String> unionTypeVarMap;
+    private final String unionVarConstantsClass;
+    private int constantIndex = 0;
+    private JvmTypeGen jvmTypeGen;
+    private ClassWriter cw;
+    private MethodVisitor mv;
+    private int memberCountInMethod;
+    private int methodCount;
+    /**
+     * Stack keeps track of recursion in union types. The method creation is performed only if recursion is completed.
+     */
+    private final Deque<BUnionType> stack;
+
+    private static final int MAX_MEMBERS_PER_METHOD = 100;
+
+    public JvmUnionTypeConstantsGen(PackageID packageID) {
+        unionTypeVarMap = new ConcurrentSkipListMap<>(this::checkUnionEqualityInInts);
+        unionVarConstantsClass = JvmCodeGenUtil.getModuleLevelClassName(
+                packageID, JvmConstants.BUNION_TYPE_CONSTANT_CLASS_NAME);
+        stack = new LinkedList<>();
+        generateClassInit();
```

Comment:
The method names `visitMethod` & `generateClassInit` seems to be generic. Shall we have something like `visitUnionInitMethod()` & `genUnionConstantsClassInit()`